src/Symfony/Component/Console/Application.php

```diff
@@ -1005,10 +1014,6 @@ protected function doRunCommand(Command $command, InputInterface $input, OutputI
                     });
                 }
             }
-
-            foreach ($command->getSubscribedSignals() as $signal) {
```

Comment:
Now it's not possible to register to a signal if the main application hasn't already registered some signals or I missed something?

Reply:
One, or both, of us might be :thinking: :woman_shrugging: :smiley_cat: 

The loop & call to `$this->signalRegistry->register($signal, [$command, 'handleSignal']);` is still present above (on lines 996-998). 

What I have been struggling with is how to test exactly the scenario you describe … As I will happily include said test, and it can contribute to outcome of this debate/question. Help/suggestions most welcome!

Reply:
@lyrixx Can you have a look at this one?

Reply:
I tested many scenario and everything seems OK. Sorry for the very long delay!